org.eclipse.lsp4j.jsonrpc/src/test/java/org/eclipse/lsp4j/jsonrpc/test/json/EitherTest.java

```diff
@@ -110,16 +111,33 @@ public void testEqualsFalseWithNonNull() {
 
 	@Test
 	public void testMap() {
-		Either<byte[], String> either =Either.forLeft(new byte[] { 'f', 'o', 'o'});
+		Either<char[], String> either = Either.forLeft(new char[] { 'f', 'o', 'o' });
 		assertEquals("foo", either.map(
 				String::new,
 				String::toUpperCase));
-		either = Either.forRight("barlol");
-		assertEquals("BARLOL", either.map(
+		either = Either.forRight("bar");
+		assertEquals("BAR", either.map(
 				String::new,
 				String::toUpperCase));
 	}
 
+	@Test
+	public void testMapEither3() {
+		Either3<String, Integer, Boolean> either3;
+		Function<String, String> mapFirst = s -> s.toUpperCase();
```

Comment:
Only the most minor of style comments - the `testMap` above uses `String::toUppercase` and this code uses `s -> s.toUpperCase()`

Reply:
Thank you for the review, Jonah! Yes, not using a method reference here is (kind of) intentional, to better match the following definitions of `mapSecond` and `mapThird`, where method references unfortunately cannot be used because of the ambiguity reported by the compiler. So, I would suggest to either leave this code as it is or change the code in `testMap` to also not use method references for more consistency (I slightly prefer the first alternative). WDYT?

Reply:
Leave the code as is.

Reply:
OK. Thanks!